scanrepository/scanrepository.go

```diff
@@ -80,16 +85,34 @@ func (cfp *ScanRepositoryCmd) scanAndFixBranch(repository *utils.Repository) (er
 		}
 		err = errors.Join(err, restoreBaseDir(), fileutils.RemoveTempDir(clonedRepoDir))
 	}()
-	if err = cfp.scanDetails.CreateMultiScanIdForScans(); err != nil {
-		return err
+
+	cfp.analyticsService = utils.AddAnalyticsGeneralEvent(cfp.scanDetails.XscGitInfoContext, cfp.scanDetails.ServerDetails, analyticsScanRepositoryScanType)
+	defer func() {
+		if err != nil && cfp.analyticsService.ShouldReportEvents() {
+			cfp.analyticsService.UpdateXscAnalyticsGeneralEventFinalizeStatus(xscservices.Failed)
+			cfp.analyticsService.UpdateGeneralEvent(cfp.analyticsService.FinalizeEvent())
+		}
+	}()
+
+	// If MSI exists we always need to report events
+	if cfp.analyticsService.GetMsi() != "" {
+		// MSI is passed to XrayGraphScanParams, so it can be later used by other analytics events in the scan phase
+		cfp.scanDetails.XrayGraphScanParams.MultiScanId = cfp.analyticsService.GetMsi()
 	}
+
 	for i := range repository.Projects {
 		cfp.scanDetails.Project = &repository.Projects[i]
 		cfp.projectTech = []coreutils.Technology{}
 		if err = cfp.scanAndFixProject(repository); err != nil {
 			return
 		}
 	}
+
+	if cfp.analyticsService.ShouldReportEvents() {
+		cfp.analyticsService.UpdateXscAnalyticsGeneralEventFinalizeWithTotalScanDuration()
+		cfp.analyticsService.UpdateXscAnalyticsGeneralEventFinalizeStatus(xscservices.Completed)
+		cfp.analyticsService.UpdateGeneralEvent(cfp.analyticsService.FinalizeEvent())
+	}
```

Comment:
```suggestion
	cfp.analyticsService = utils.AddAnalyticsGeneralEvent(cfp.scanDetails.XscGitInfoContext, cfp.scanDetails.ServerDetails, analyticsScanRepositoryScanType)
	defer func() {
        if !cfp.analyticsService.ShouldReportEvents() {
			return
        }
        cfp.analyticsService.UpdateXscAnalyticsGeneralEventFinalizeStatus(xscservices.Completed)
		if err != nil {
			cfp.analyticsService.UpdateXscAnalyticsGeneralEventFinalizeStatus(xscservices.Failed)
		}
        cfp.analyticsService.UpdateGeneralEvent(cfp.analyticsService.FinalizeEvent())
	}()

	// If MSI exists we always need to report events
	if cfp.analyticsService.GetMsi() != "" {
		// MSI is passed to XrayGraphScanParams, so it can be later used by other analytics events in the scan phase
		cfp.scanDetails.XrayGraphScanParams.MultiScanId = cfp.analyticsService.GetMsi()
	}

	for i := range repository.Projects {
		cfp.scanDetails.Project = &repository.Projects[i]
		cfp.projectTech = []coreutils.Technology{}
		if err = cfp.scanAndFixProject(repository); err != nil {
			return
		}
	}
```
* move the defer for `UpdateGeneralEvent` to be the first defer at the func (so it will be the last to run when done), move it above the defer for the restore that could result with error and change the status.
* no need for duplicate code (calling for the update)